lib/LTP/utils.pm

```diff
@@ -352,4 +357,67 @@ sub parse_runfiles {
     }
 }
 
+sub add_ltp_repo {
+    my $repo = get_var('LTP_REPOSITORY');
+
+    if (!$repo) {
+        if (is_sle || is_transactional) {
+            add_qa_head_repo;
+            return;
+        }
+
+        # ltp for leap15.2 is available only x86_64
+        if (is_leap('15.4+')) {
+            $repo = get_var('VERSION');
+        } elsif ((is_leap('=15.2') && is_x86_64) || is_leap('15.3+')) {
+            $repo = sprintf("openSUSE_Leap_%s", get_var('VERSION'));
+        } elsif (is_tumbleweed) {
+            $repo = "openSUSE_Factory";
+            $repo = "openSUSE_Factory_ARM" if (is_aarch64() || is_arm());
+            $repo = "openSUSE_Factory_PowerPC" if is_ppc64le();
+            $repo = "openSUSE_Factory_zSystems" if is_s390x();
+        } else {
+            die sprintf("Unexpected combination of version (%s) and architecture (%s) used", get_var('VERSION'), get_var('ARCH'));
+        }
+        $repo = "https://download.opensuse.org/repositories/benchmark:/ltp:/devel/$repo/";
+    }
+
+    zypper_ar($repo, name => 'ltp_repo');
+}
+
+sub get_default_pkg {
+    my @packages;
+
+    if (is_sle && is_released) {
+        push @packages, 'ltp-stable';
+        push @packages, 'ltp-stable-32bit' if is_x86_64;
+    } else {
+        push @packages, 'ltp';
+        push @packages, 'ltp-32bit' if is_x86_64 && !is_jeos;
+    }
+
+    return join(' ', @packages);
+}
+
+sub install_from_repo {
+    my @pkgs = split(/\s* \s*/, get_var('LTP_PKG', get_default_pkg));
+
+    if (is_transactional) {
+        assert_script_run("transactional-update -n -c pkg install " . join(' ', @pkgs), 180);
+    } else {
+        zypper_call("in --recommends " . join(' ', @pkgs));
+    }
+
+    my $run_cmd = is_transactional ? 'transactional-update -c -d --quiet run' : '';
+    for my $pkg (@pkgs) {
+        my $want_32bit = want_ltp_32bit($pkg);
+
+        record_info("LTP pkg: $pkg", script_output("$run_cmd rpm -qi $pkg | tee "
+                  . get_ltp_version_file($want_32bit)));
```

Comment:
Do we really need `transactional-update` to just query information about a package? I would expect this to work on a read-only filesystem. It looks like the `trup_install` function you call later also avoids this.

Reply:
It's used only when system is transactional

Reply:
My point was: Even when we are running a transactional system it is likely not necessary to use `transactional-update` for this read-only rpm-command. The function `trup_install` also does a similar rpm call upfront without using `transactional-update` so it is likely possible to avoid it here as well.

Reply:
You are right. The `add_ltp_repo` subroutine is needed by the kirk integration code and I prefer to move it inside `LTP::utils` without changing it, since the subject of this pull request is a feature unrelated with aesthetics or other optimizations in the LTP repo handling code.

Reply:
@Martchus feel free to implement it as a separate effort afterwards.